src/core/src/Axis.cpp

```diff
@@ -128,6 +152,31 @@ namespace iDynTree
 
         return derivative_nonRotated_T_rotated;
     }
+    
+    TransformDerivative Axis::getTranslationTransformDerivative(const double dist) const
```

Comment:
The failure in the tests highlights a problem in this function. Are you sure that it is implementing what is described in its documentation, i.e. : "Get the derivative of the getTranslationTransform function with respect to the dist argument" ?